ws-client/src/transport.rs

```diff
@@ -294,9 +295,14 @@ impl Target {
 			url.host_str().map(ToOwned::to_owned).ok_or_else(|| WsHandshakeError::Url("No host in URL".into()))?;
 		let port = url.port_or_known_default().ok_or_else(|| WsHandshakeError::Url("No port number in URL".into()))?;
 		let host_header = format!("{}:{}", host, port);
+		let mut path = url.path().to_owned();
+		if let Some(query) = url.query() {
+			path.push('?');
+			path.push_str(query);
+		}
```

Comment:
This looks sane to me (without a wider context of how the path `String` is used elsewhere)!

I was wondering whether percent-encoded URLs would end up being double percent-encoded when returned (because Url::query percent-encodes) but I did this:

```
use url::Url;

fn main() {
    let foo = Url::parse("ws://localhost/foo?bar=hello%20world").unwrap();
    println!("{:?}", foo.query());
}
```

in the rust playground and the output is "bar=hello%20world" as I'd hope for :)